From 9387d2f56981b2786a70cfded94b0ad2012a6dcc Mon Sep 17 00:00:00 2001
From: Thierry Reding <thierry.reding@avionic-design.de>
Date: Tue, 24 Jul 2012 16:24:41 +0200
Subject: [PATCH] freetype: Fix a possible buffer corruption

Memory associated with a FcPattern is freed when the pattern is
destroyed, so copying the font filename needs to be done before
the call to FcPatternDestroy.

Signed-off-by: Thierry Reding <thierry.reding@avionic-design.de>
---
 modules/text_renderer/freetype.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/modules/text_renderer/freetype.c b/modules/text_renderer/freetype.c
index b862bb4..4e82f3b 100644
--- a/modules/text_renderer/freetype.c
+++ b/modules/text_renderer/freetype.c
@@ -505,6 +505,7 @@ static char* FontConfig_Select( FcConfig* config, const char* family,
     FcPattern *pat, *p_pat;
     FcChar8* val_s;
     FcBool val_b;
+    char *ret;
 
     /* Create a pattern and fills it */
     pat = FcPatternCreate();
@@ -567,8 +568,9 @@ static char* FontConfig_Select( FcConfig* config, const char* family,
         return NULL;
     }
 
+    ret = strdup( (const char*)val_s );
     FcPatternDestroy( p_pat );
-    return strdup( (const char*)val_s );
+    return ret;
 }
 #endif
 
-- 
1.7.11.2

